contrib/kafka/cmd/dispatcher/main.go

```diff
@@ -55,16 +59,30 @@ func main() {
 		logger.Fatal("Unable to add kafkaDispatcher", zap.Error(err))
 	}
 
-	if err := v1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+	if err = v1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
 		logger.Fatal("Unable to add scheme for eventing apis.", zap.Error(err))
 	}
 
-	if err := channelwatcher.New(mgr, logger, channelwatcher.UpdateConfigWatchHandler(kafkaDispatcher.UpdateConfig, shouldWatch)); err != nil {
+	// Zipkin tracing.
+	kc := kubernetes.NewForConfigOrDie(mgr.GetConfig())
+	configMapWatcher := configmap.NewInformedWatcher(kc, system.Namespace())
```

Comment:
NewInformedWatcher returns a *InformedWatcher which is being sent to SetupDynamicZipkinPublishing as an interface. Code inside SetupDynamicZipkinPublishing never checks for nil. Doesn't this has a potential to crash the app?

Reply:
Yes. We tend to play a bit fast and loose with pointers.

As this is `main` and failure will occur almost immediately, I am fine with it this way. But, if you prefer, I can add the check. Would you like me to add it?